db/inmemory/client.go

```diff
@@ -24,22 +25,19 @@ func (c *client) AddUser(user *domain.User) (string, error){
 	c.ds[user.ID]=user
 	return user.ID,nil
 }
-func (c *client) ListUsers(queryName *string, queryLimit *int32) ([]*domain.User,error){
+func (c *client) ListUsers(queryNameMap map[string]interface{}, queryLimit int32) ([]*domain.User,error){
```

Comment:
instead of `queryNameMap` use `filters`
and `queryLimit` will be `limit` only

Reply:
Done ! Please check